apps/api/v2/src/ee/calendars/services/apple-calendar.service.ts

```diff
@@ -61,26 +62,70 @@ export class AppleCalendarService implements CredentialSyncCalendarApp {
     if (username.length <= 1 || password.length <= 1)
       throw new BadRequestException(`Username or password cannot be empty`);
 
-    const data = {
-      type: APPLE_CALENDAR_TYPE,
-      key: symmetricEncrypt(
-        JSON.stringify({ username, password }),
-        process.env.CALENDSO_ENCRYPTION_KEY || ""
-      ),
-      userId: userId,
-      teamId: null,
-      appId: APPLE_CALENDAR_ID,
-      invalid: false,
-    };
+    const existingAppleCalendarCredentials = await this.credentialRepository.getAllUserCredentialsByTypeAndId(
+      APPLE_CALENDAR_TYPE,
+      userId
+    );
+
+    let hasMatchingUsernameAndPassword = false;
+
+    if (existingAppleCalendarCredentials.length > 0) {
+      const hasCalendarWithGivenCredentials = existingAppleCalendarCredentials.find(
+        (calendarCredential: Credential) => {
+          const decryptedKey = JSON.parse(
+            symmetricDecrypt(calendarCredential.key as string, process.env.CALENDSO_ENCRYPTION_KEY || "")
+          );
+
+          if (decryptedKey.username === username) {
+            if (decryptedKey.password === password) {
+              hasMatchingUsernameAndPassword = true;
+            }
+
+            return true;
+          }
+        }
+      );
+
+      if (!!hasCalendarWithGivenCredentials && hasMatchingUsernameAndPassword) {
+        return {
+          status: SUCCESS_STATUS,
+        };
+      }
+
+      if (!!hasCalendarWithGivenCredentials && !hasMatchingUsernameAndPassword) {
```

Comment:
For this case, there is the same user but with a different password. Now assuming that the user's app specific password got expired and they're trying to update the password we do an upsert where we just update they're credential key.